Python 3.14: RuntimeError: There is no current event loop in thread 'MainThread'

##### Description:
In Fedora Linux, we have tried to build urwid with Python 3.14 and several tests failed with: ` RuntimeError: There is no current event loop in thread 'MainThread'`.

```
__________________ AsyncioEventLoopTest.test_run_in_executor ___________________

self = <tests.test_event_loops.AsyncioEventLoopTest testMethod=test_run_in_executor>

    def setUp(self):
>       self.evl = urwid.AsyncioEventLoop()

tests/test_event_loops.py:216: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
urwid/event_loop/asyncio_loop.py:83: in __init__
    self._loop = asyncio.get_event_loop()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <asyncio.unix_events._UnixDefaultEventLoopPolicy object at 0x7f152f488050>

    def get_event_loop(self):
        """Get the event loop for the current context.
    
        Returns an instance of EventLoop or raises an exception.
        """
        if self._local._loop is None:
>           raise RuntimeError('There is no current event loop in thread %r.'
                               % threading.current_thread().name)
E           RuntimeError: There is no current event loop in thread 'MainThread'.

/usr/lib64/python3.14/asyncio/events.py:681: RuntimeError
=========================== short test summary info ============================
FAILED tests/test_event_loops.py::AsyncioEventLoopTest::test_coroutine_error - RuntimeError: There is no current event loop in thread 'MainThread'.
FAILED tests/test_event_loops.py::AsyncioEventLoopTest::test_error - RuntimeError: There is no current event loop in thread 'MainThread'.
FAILED tests/test_event_loops.py::AsyncioEventLoopTest::test_event_loop - RuntimeError: There is no current event loop in thread 'MainThread'.
FAILED tests/test_event_loops.py::AsyncioEventLoopTest::test_remove_alarm - RuntimeError: There is no current event loop in thread 'MainThread'.
FAILED tests/test_event_loops.py::AsyncioEventLoopTest::test_remove_watch_file - RuntimeError: There is no current event loop in thread 'MainThread'.
FAILED tests/test_event_loops.py::AsyncioEventLoopTest::test_run - RuntimeError: There is no current event loop in thread 'MainThread'.
FAILED tests/test_event_loops.py::AsyncioEventLoopTest::test_run_in_executor - RuntimeError: There is no current event loop in thread 'MainThread'.
=========== 7 failed, 285 passed, 28 skipped, 221 warnings in 4.98s ============
```

Link to the original bug report created by the Fedora Python maintainer: https://bugzilla.redhat.com/show_bug.cgi?id=2327984

##### Affected versions (if applicable)
- [ ] `master` branch (specify commit)
- [x] Latest stable version from `pypi`
- [ ] Other (specify source)

##### Steps to reproduce (if applicable)
Launch tests with Python 3.14.0a2

##### Expected/actual outcome
All tests pass

--- a/tests/test_event_loops.py
+++ b/tests/test_event_loops.py
@@ -213,7 +213,17 @@ class SelectEventLoopTest(unittest.TestCase, EventLoopTestMixin):
 @unittest.skipIf(IS_WINDOWS, "Windows is temporary not supported by AsyncioEventLoop.")
 class AsyncioEventLoopTest(unittest.TestCase, EventLoopTestMixin):
     def setUp(self):
-        self.evl = urwid.AsyncioEventLoop()
+        if sys.version_info[:2] < (3, 11):
+            self.loop = asyncio.get_event_loop_policy().get_event_loop()
+            self.evl_runner = None
+        else:
+            self.evl_runner = asyncio.Runner(loop_factory=asyncio.SelectorEventLoop)
+            self.loop = self.evl_runner.get_loop()
+        self.evl = urwid.AsyncioEventLoop(loop=self.loop)
+
+    def tearDown(self):
+        if self.evl_runner is not None:
+            self.evl_runner.close()
 
     _expected_idle_handle = None
 
@@ -233,7 +243,7 @@ class AsyncioEventLoopTest(unittest.TestCase, EventLoopTestMixin):
             result = 1 / 0  # Simulate error in coroutine
             return result
 
-        asyncio.ensure_future(error_coro(), loop=asyncio.get_event_loop_policy().get_event_loop())
+        asyncio.ensure_future(error_coro(), loop=self.loop)
         self.assertRaises(ZeroDivisionError, evl.run)
 
 

--- a/urwid/canvas.py
+++ b/urwid/canvas.py
@@ -1170,7 +1170,7 @@ def cview_trim_rows(cv, rows: int):
 
 
 def cview_trim_top(cv, trim: int):
-    return (cv[0], trim + cv[1], cv[2], cv[3] - trim) + cv[4:]
+    return (cv[0], trim + cv[1], cv[2], cv[3] - trim, *cv[4:])
 
 
 def cview_trim_left(cv, trim: int):

--- a/urwid/event_loop/asyncio_loop.py
+++ b/urwid/event_loop/asyncio_loop.py
@@ -41,7 +41,6 @@ if typing.TYPE_CHECKING:
     _T = typing.TypeVar("_T")
 
 __all__ = ("AsyncioEventLoop",)
-IS_WINDOWS = sys.platform == "win32"
 
 
 class AsyncioEventLoop(EventLoop):
@@ -67,20 +66,37 @@ class AsyncioEventLoop(EventLoop):
     def __init__(self, *, loop: asyncio.AbstractEventLoop | None = None, **kwargs) -> None:
         super().__init__()
         self.logger = logging.getLogger(__name__).getChild(self.__class__.__name__)
-        if loop:
-            self._loop: asyncio.AbstractEventLoop = loop
+
+        if sys.version_info[:2] < (3, 11):
             self._event_loop_policy_altered: bool = False
             self._original_event_loop_policy: asyncio.AbstractEventLoopPolicy | None = None
-        else:
-            self._original_event_loop_policy = asyncio.get_event_loop_policy()
-            if IS_WINDOWS and not isinstance(self._original_event_loop_policy, asyncio.WindowsSelectorEventLoopPolicy):
-                self.logger.debug("Set WindowsSelectorEventLoopPolicy as asyncio event loop policy")
-                asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
-                self._event_loop_policy_altered = True
-            else:
-                self._event_loop_policy_altered = False
 
-            self._loop = asyncio.get_event_loop()
+            if loop:
+                self._loop: asyncio.AbstractEventLoop = loop
+            else:
+                self._original_event_loop_policy = asyncio.get_event_loop_policy()
+                if sys.platform == "win32" and not isinstance(
+                    self._original_event_loop_policy, asyncio.WindowsSelectorEventLoopPolicy
+                ):
+                    self.logger.debug("Set WindowsSelectorEventLoopPolicy as asyncio event loop policy")
+                    asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
+                    self._event_loop_policy_altered = True
+                else:
+                    self._event_loop_policy_altered = False
+
+                self._loop = asyncio.get_event_loop()
+
+        else:
+            self._runner: asyncio.Runner | None = None
+
+            if loop:
+                self._loop: asyncio.AbstractEventLoop = loop
+            else:
+                try:
+                    self._loop = asyncio.get_running_loop()
+                except RuntimeError:
+                    self._runner = asyncio.Runner(loop_factory=asyncio.SelectorEventLoop)
+                    self._loop = self._runner.get_loop()
 
         self._exc: BaseException | None = None
 
@@ -89,8 +105,11 @@ class AsyncioEventLoop(EventLoop):
         self._idle_callbacks: dict[int, Callable[[], typing.Any]] = {}
 
     def __del__(self) -> None:
-        if self._event_loop_policy_altered:
-            asyncio.set_event_loop_policy(self._original_event_loop_policy)  # Restore default event loop policy
+        if sys.version_info[:2] < (3, 11):
+            if self._event_loop_policy_altered:
+                asyncio.set_event_loop_policy(self._original_event_loop_policy)  # Restore default event loop policy
+        elif self._runner is not None:
+            self._runner.close()
 
     def _also_call_idle(self, callback: Callable[_Spec, _T]) -> Callable[_Spec, _T]:
         """

